plugins/generator-1.18.2/forge-1.18.2/templates/dispensebehaviour.java.ftl

```diff
@@ -35,34 +35,39 @@
 
 package ${package}.item.extension;
 
+<#compress>
 @Mod.EventBusSubscriber(bus = Mod.EventBusSubscriber.Bus.MOD) public class ${name}ItemExtension {
     	@SubscribeEvent public static void init(FMLCommonSetupEvent event) {
     		event.enqueueWork(() -> DispenserBlock.registerBehavior(${mappedMCItemToItem(data.item)}, new OptionalDispenseItemBehavior() {
     			public ItemStack execute(BlockSource blockSource, ItemStack stack) {
+    				<#assign hasSuccessCondition = hasProcedure(data.dispenseSuccessCondition)>
     				ItemStack itemstack = stack.copy();
     				Level world = blockSource.getLevel();
     				Direction direction = blockSource.getBlockState().getValue(DispenserBlock.FACING);
     				int x = blockSource.getPos().getX();
     				int y = blockSource.getPos().getY();
     				int z = blockSource.getPos().getZ();
 
+    				<#if hasSuccessCondition>
     				this.setSuccess(<@procedureOBJToConditionCode data.dispenseSuccessCondition/>);
+    				</#if>
 
     				<#if hasProcedure(data.dispenseResultItemstack)>
     					boolean success = this.isSuccess();
     					<#if hasReturnValueOf(data.dispenseResultItemstack, "itemstack")>
-
-    						return <@procedureOBJToItemstackCode data.dispenseResultItemstack/>;
+    						return <@procedureOBJToItemstackCode data.dispenseResultItemstack, false/>;
     					<#else>
     						<@procedureOBJToCode data.dispenseResultItemstack/>
-    						if(success) itemstack.shrink(1);
+    						<#if hasSuccessCondition>if(success)</#if>

```

Comment:
`boolean success = this.isSuccess();` is only used here now, so I think it could be inlined

Reply:
There's a procedure that uses this variable ~afair~.

Reply:
`success` is also passed as a dependency to the itemstack procedure, so it can't be inlined

Reply:
Yes, that one.